lib/rspec/matchers/built_in/change.rb

```diff
@@ -87,8 +87,8 @@ def initialize(receiver=nil, message=nil, &block)
         end
 
         def raise_block_syntax_error
-          raise SyntaxError, "The block passed to the `change` matcher must " \
-            "use `{ ... }` instead of do/end"
+          raise SyntaxError, "Block passed to the wrong method. Perhaps you " \
```

Comment:
If I run
```ruby
expect{ subject }.to(change{}){ some_query.count }.by(1)
```
it throws the error. In that case there *is* a block given to the `change` matcher. The check that's done in the code is a positive `block_given?` check through the `to` method (and I think some others?). I'd like to write `Block passed to #{__method__} method` except that I haven't figured out how to do that yet.